### PR TITLE
feat: handle a few cases that are now handled in the JS parser

### PR DESCRIPTION
**Pull request by rhamzeh**
## Description

This updates the go parser in language to pass the tests (Note: I have not added the validation tests yet, this is just for the syntax tests)

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/go/transformer/dsltojson.go [hunk 1]
```diff
@@ -66,37 +71,47 @@ func (l *openFgaDslListener) ExitTypeDef(ctx *parser.TypeDefContext) {
...
 
-func (l *openFgaDslListener) ExitRelationDeclaration(ctx *parser.RelationDeclarationContext) {
+func (l *OpenFgaDslListener) ExitRelationDeclaration(ctx *parser.RelationDeclarationContext) {
+	if ctx.RelationName() == nil {
+		return
+	}
+
 	relationName := ctx.RelationName().GetText()
 
+	if l.currentRelation.Rewrites == nil || len(l.currentRelation.Rewrites) == 0 {
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson.go:88]: Do we need to check for whether l.currentRelation is null?

File: pkg/go/transformer/dsltojson.go [hunk 2]
```diff
@@ -66,37 +71,47 @@ func (l *openFgaDslListener) ExitTypeDef(ctx *parser.TypeDefContext) {
...
+		return
+	}
+
+	var relationDef *pb.Userset
+
 	if len(l.currentRelation.Rewrites) == 1 {
-		l.currentTypeDef.Relations[relationName] = l.currentRelation.Rewrites[0]
+		relationDef = l.currentRelation.Rewrites[0]
 	} else {
 		if l.currentRelation.Operator == RELATION_DEFINITION_OPERATOR_OR {
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson.go:97]: should we use switch to be more efficient?

File: pkg/go/transformer/dsltojson.go [hunk 3]
```diff
@@ -107,24 +122,30 @@ func (l *openFgaDslListener) ExitRelationDeclaration(ctx *parser.RelationDeclara
...
 	}
 
-	directlyRelatedUserTypes := l.currentRelation.TypeInfo.GetDirectlyRelatedUserTypes()
+	if relationDef != nil {
+		if l.currentTypeDef.Relations[relationName] != nil {
+			ctx.GetParser().NotifyErrorListeners(fmt.Sprintf("%s is already defined in %s", relationName, l.currentTypeDef.Type), ctx.GetParser().GetCurrentToken(), nil)
+		}
 
-	l.currentTypeDef.Metadata.Relations[relationName] = &pb.RelationMetadata{DirectlyRelatedUserTypes: directlyRelatedUserTypes}
+		l.currentTypeDef.Relations[relationName] = relationDef
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson.go:130]: Need to check whether l.currentTypeDef is not nil?

File: pkg/go/transformer/dsltojson.go [hunk 4]
```diff
@@ -176,68 +197,113 @@ func (l *openFgaDslListener) ExitRelationDefRelationOnRelatedObject(
...
+
+	pluralS := ""
+	if len(errors) > 1 {
+		pluralS = "s"
+	}
+
+	errorsString := []string{}
+	for index := 0; index < len(errors); index++ {
+		errorsString = append(errorsString, errors[index].Error())
+	}
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson.go:243]:
```suggestion
        for _, item := range errors {
                errorsString = append(errorsString, item.Error())
        }
```

**Review by adriantam — CHANGES_REQUESTED** [review]

File: pkg/go/transformer/dsltojson.go (hunk 1 above)
Comment (rhamzeh) [reply pkg/go/transformer/dsltojson.go:88]: Already initialized here: https://github.com/openfga/language/blob/50b003d312da3fdb52812a562ea22292bc186e6e/pkg/go/transformer/dsltojson.go#L74

File: pkg/go/transformer/dsltojson.go (hunk 3 above)
Comment (rhamzeh) [reply pkg/go/transformer/dsltojson.go:130]: Already initialized here: https://github.com/openfga/language/blob/50b003d312da3fdb52812a562ea22292bc186e6e/pkg/go/transformer/dsltojson.go#L48

**Review by adriantam — DISMISSED** [review]

**Review by adriantam — APPROVED** [review]

**Pull request merged**